### PR TITLE
Add FromGeographic() method to EcefLocation

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
imodel-native-internal: https://github.com/iTwin/imodel-native-internal/pull/505

This PR is required for the above-linked change. This PR adds a new method titled FromGeographic() to the EcefLocation class. This allows a user to create an EcefLocation from lat-long coordinates on an ECEF globe, without needing any GCS.

### CODE REVIEW

**Review by pmconne — COMMENTED** [review]
tests

File: iModelCore/iModelPlatform/PublicAPI/DgnPlatform/DgnDbTables.h [hunk 1]
```diff
@@ -581,6 +581,11 @@ struct EcefLocation
     // Constructor to compute ECEF location from an cartographic origin
     DGNPLATFORM_EXPORT EcefLocation(GeoPointCR originLL, DPoint3dCR originIn, GeoCoordinates::BaseGCSCP GCS);
     DGNPLATFORM_EXPORT Transform ComputeTransform();
+    //! Convert a geographic point to an ECEF (Earth-Centered, Earth-Fixed) location.
```
Comment (mathieu-lemuzic) [inline iModelCore/iModelPlatform/PublicAPI/DgnPlatform/DgnDbTables.h:584]: should we cover this with a small unit test ?

Comment (markschlosseratbentley) [reply iModelCore/iModelPlatform/PublicAPI/DgnPlatform/DgnDbTables.h:584]: Yeah, as Paul suggested on Friday --- working on it.

Comment (markschlosseratbentley) [reply iModelCore/iModelPlatform/PublicAPI/DgnPlatform/DgnDbTables.h:584]: pushed

**Review by pmconne — APPROVED** [review]

**Pull request merged**